src/main/java/com/atlassian/db/replica/spi/Chief.java

```diff
@@ -0,0 +1,14 @@
+package com.atlassian.db.replica.spi;
+
+import com.atlassian.db.replica.internal.RouteDecisionBuilder;
+
+import java.sql.Connection;
+import java.util.Collection;
+import java.util.function.Supplier;
+
+/**
+ * Can override the automatic decision to make a final call where the query should go.
```

Comment:
Could we clarify the lifecycle of a `Chief`? Is it reused across multiple database calls? Multiple connections? Multiple pools?

We can use the [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) MUST, SHOULD, MAY keywords.

Reply:
The life cycle is up to the customer. For example, a customer can inject a new instance every DualConnection or use a singleton.

Reply:
MAY the customer choose to have a `new` Chief for every `DatabaseCall` or MUST every Chief impl handle multiple calls?
Is `Chief` lifecycle equivalent to `DualConnection` lifecycle? Does it have to be `@ThreadSafe` or is it ok to be `@NotThreadSafe`?
If reusing `Chief`s across DB pool is recommended, then they SHOULD do it. If it's recommended to avoid that, then they SHOULD NOT do it. If it's ok either way, they MAY do it. OTOH the stance on this lifecycle probably SHOULD be documented in the `DuallConnection.Builder` injection site rather than in the `Chief` class itself.